pkg/sdn/plugin/node_iptables.go

```diff
@@ -78,24 +78,49 @@ func (n *NodeIPTables) syncIPTableRules() error {
 	}()
 	glog.V(3).Infof("Syncing openshift iptables rules")
 
-	rules := n.getStaticNodeIPTablesRules()
-	for _, rule := range rules {
-		_, err := n.ipt.EnsureRule(iptables.Prepend, iptables.Table(rule.table), iptables.Chain(rule.chain), rule.args...)
+	for _, rule := range n.getStaticNodeIPTablesChains() {
+		chain := rule.args[len(rule.args)-1]
+		_, err := n.ipt.EnsureChain(iptables.Table(rule.table), iptables.Chain(chain))
+		if err != nil {
+			return fmt.Errorf("failed to ensure chain %s exists: %v", chain, err)
+		}
+		if flush {
+			err = n.ipt.FlushChain(iptables.Table(rule.table), iptables.Chain(chain))
+			if err != nil {
+				return fmt.Errorf("failed to flush chain %s: %v", chain, err)
+			}
+		}
+		_, err = n.ipt.EnsureRule(iptables.Prepend, iptables.Table(rule.table), iptables.Chain(rule.chain), rule.args...)
+		if err != nil {
+			return fmt.Errorf("failed to ensure rule %v exists: %v", rule, err)
+		}
+	}
+
+	for _, rule := range n.getStaticNodeIPTablesRules() {
+		_, err := n.ipt.EnsureRule(iptables.Append, iptables.Table(rule.table), iptables.Chain(rule.chain), rule.args...)
 		if err != nil {
 			return fmt.Errorf("failed to ensure rule %v exists: %v", rule, err)
```

Comment:
doesn't really matter; it's the same error really, and the error message includes the rule itself (and the line number) so it will be clear what exactly failed